lib/index.ts

```diff
@@ -201,17 +201,25 @@ export async function createDriver(options: {extraArgs?: string[]} = {}): Promis
   return newDriver;
 }
 
+let _driverCreationPromise: Promise<WebDriver> | undefined = undefined;
+
 // Start up the webdriver and serve files that its browser will see.
 export async function beforeMochaWebdriverTests(this: Mocha.Context) {
   // If this has already been called, there's nothing to do.
-  if (_driver) { return; }
+  if (_driver || _driverCreationPromise) { return; }
 
   this.timeout(20000);      // Set a longer default timeout.
 
   // Add stack trace enhancement (no-op if MOCHA_WEBDRIVER_STACKTRACES isn't set).
   stackWrapDriverMethods();
 
-  setDriver(await createDriver());
+  try {
+    _driverCreationPromise = createDriver();
+    setDriver(await _driverCreationPromise);
+  } catch(e) {
+    _driverCreationPromise = undefined;
```

Comment:
Why the try-catch here? If there's an exception, `_driverCreationPromise` will be `undefined` anyway.

Reply:
As George mentioned, we'd get a rejected promise which is truthy, which would prevent this function from being able to be executed again on failure.